File: clients/Python-PyPy/model/move_action.py
from .vec2_int import Vec2Int
class MoveAction:
    def __init__(self, target, find_closest_position, break_through):
        self.target = target
        self.find_closest_position = find_closest_position
        self.break_through = break_through
    @staticmethod
    def read_from(stream):
        target = Vec2Int.read_from(stream)
        find_closest_position = stream.read_bool()
        break_through = stream.read_bool()
        return MoveAction(target, find_closest_position, break_through)
    def write_to(self, stream):
        self.target.write_to(stream)
        stream.write_bool(self.find_closest_position)
        stream.write_bool(self.break_through)
    def __repr__(self):
        return "MoveAction(" + \
            repr(self.target) + "," + \
            repr(self.find_closest_position) + "," + \
            repr(self.break_through) + \
            ")"

File: clients/Java-GraalVM-native/src/main/java/model/MoveAction.java
package model;

import util.StreamUtil;

public class MoveAction {
    private model.Vec2Int target;
    public model.Vec2Int getTarget() { return target; }
    public void setTarget(model.Vec2Int target) { this.target = target; }
    private boolean findClosestPosition;
    public boolean isFindClosestPosition() { return findClosestPosition; }
    public void setFindClosestPosition(boolean findClosestPosition) { this.findClosestPosition = findClosestPosition; }
    private boolean breakThrough;
    public boolean isBreakThrough() { return breakThrough; }
    public void setBreakThrough(boolean breakThrough) { this.breakThrough = breakThrough; }
    public MoveAction() {}
    public MoveAction(model.Vec2Int target, boolean findClosestPosition, boolean breakThrough) {
        this.target = target;
        this.findClosestPosition = findClosestPosition;
        this.breakThrough = breakThrough;
    }
    public static MoveAction readFrom(java.io.InputStream stream) throws java.io.IOException {
        MoveAction result = new MoveAction();
        result.target = model.Vec2Int.readFrom(stream);
        result.findClosestPosition = StreamUtil.readBoolean(stream);
        result.breakThrough = StreamUtil.readBoolean(stream);
        return result;
    }
    public void writeTo(java.io.OutputStream stream) throws java.io.IOException {
        target.writeTo(stream);
        StreamUtil.writeBoolean(stream, findClosestPosition);
        StreamUtil.writeBoolean(stream, breakThrough);
    }
}
